app/pb_kits/playbook/pb_checkbox/checkbox.rb

```diff
@@ -2,93 +2,51 @@
 
 module Playbook
   module PbCheckbox
-    class Checkbox < Playbook::PbKit::Base
-      PROPS = %i[configured_classname
-                 configured_data
-                 configured_id
-                 configured_dark
-                 configured_text
-                configured_value
-              configured_name
-              configured_checked
-              configured_icon
-            ].freeze
+    class Checkbox
+      include Playbook::Props
 
-      def initialize(classname: default_configuration,
-                     data: default_configuration,
-                     id: default_configuration,
-                     dark: default_configuration,
-                     text: default_configuration,
-                   value: default_configuration,
-                 name: default_configuration,
-                 checked: default_configuration,
-                 icon: default_configuration
-               )
-        self.configured_classname = classname
-        self.configured_data = data
-        self.configured_id = id
-        self.configured_dark = dark
-        self.configured_text = text
-        self.configured_value = value
-        self.configured_name = name
-        self.configured_checked = checked
-        self.configured_icon = icon
-
-      end
+      prop :dark, type: Playbook::Props::Boolean, default: false
+      prop :text, default: ""
+      prop :value, default: ""
+      prop :name, default: ""
+      prop :checked, default: nil
+      prop :icon, default: nil
 
       def to_partial_path
         "pb_checkbox/checkbox"
       end
 
-      def icon
-        if is_set? configured_icon
+      def icon_html
```

Comment:
I don't like this much. `ApplicationController` is a constant coming from the outer world, that might not even be called just that (in our componentized world it would always have some namespace, and although this would work we're basically relying on anything this constant matches).

Maybe this method could just be `icon_props` and in the view you could have a `pb_rails("icon", props: object.icon_props)` call.

Reply:
That sounds good! I'll give it a shot.